src/LocalStorageStore.ts

```diff
@@ -0,0 +1,54 @@
+import Store from "./Store";
+import MemoryStore from "./MemoryStore";
+
+/** Class that stores key/value pairs in localStorage if possible
+    but falls back to an in-memory store. */
+export default class LocalStorageStore implements Store {
+    private enabled: boolean;
+    private memoryStore: MemoryStore | null;
+    private manifestUrl: string;
+    
+    public constructor() {}
+
+    public async start(manifestUrl: string): Promise<void> {
+        this.manifestUrl = manifestUrl;
+        try {
+            // In some browsers (eg iOS Safari in private mode), 
+            // localStorage exists but throws an exception when
+            // you try to write to it.
+            window.localStorage.setItem("test", "test");
+            window.localStorage.removeItem("test");
+            this.enabled = true;
+        } catch (e) {
+            this.enabled = false;
+            this.memoryStore = new MemoryStore();
+            await this.memoryStore.start();
+        }
+        return new Promise<void>(resolve => resolve());
+    }
+
+    private getLocalStorageKey(key: string): string {
+        return this.manifestUrl + "-" + key;
+    }
+
+    public async get(key: string): Promise<string | null> {
+        let value: string | null = null;
+        if (this.enabled) {
+            value = window.localStorage.getItem(this.getLocalStorageKey(key));
+        }
+        if (this.memoryStore) {
```

Comment:
These conditionals can be cleaned up a bit when `this.enabled` is either renamed or removed (because we just will check if `this.memoryStore` is non-null).